### PR TITLE
Handle camel case geo table for localization

### LINKED ISSUE #5192: Schema labels and configuration are not being respected with new tables
**Describe the bug**
Despite each field in the schema config being assigned a label and visibility status, the forms and queries do not respect field visibility, descriptions, or captions.

This is on a local copy similar to `geo_swiss_2024_08_05` on the test panel
 
**To Reproduce**

1. Customize the schema for `collectionobjectgroup` and `collectionobjectgroupjoin`
2. Go to the query builder (`/specify/query/new/collectionobjectgroup/`) or to the form (`/specify/view/collectionobjectgroup/new/`)
3. See that the custom visibility/labels/descriptions are not being respected

**Expected behavior**
It should work as it does across other tables.

**Screenshots**
image

See that a number of these fields are hidden and my custom schema names are not being respected (`isprimary` should be `Is Primary`).

**Crash Report**
System Information:
macOS, Arc
Git SHA: https://github.com/specify/specify7/commit/9f37028b2e20f10eb9972d50f4058790b3dcca9f

[Specify 7 System Information - 2024-08-05T17_13_03.215Z.txt](https://github.com/user-attachments/files/16500712/Specify.7.System.Information.-.2024-08-05T17_13_03.215Z.txt)

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #5192

### Checklist

- [ ] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add relevant issue to release milestone

### Testing instructions

- Customize the schema for collectionobjectgroup and collectionobjectgroupjoin
- Go to the query builder (/specify/query/new/collectionobjectgroup/) or to the form (/specify/view/collectionobjectgroup/new/)
- [ ] Verify that the custom visibility/labels/descriptions are respected

### CODE REVIEW

**Review by sharadsw — APPROVED** [review]

**Comment by CarolineDenis** [comment]
NOTES: 
Might not be necessary after https://github.com/specify/specify7/pull/5221/files#diff-95fc44b05f92ce9aaaa555f15ed1b3da6e4044fb3c851ea43ae0a42561e6068f

**Review by emenslin — APPROVED** [review]
### Testing instructions

- Customize the schema for collectionobjectgroup and collectionobjectgroupjoin
- Go to the query builder (/specify/query/new/collectionobjectgroup/) or to the form (/specify/view/collectionobjectgroup/new/)
- [x] Verify that the custom visibility/labels/descriptions are respected

Looks good, all customizations seem to be respected in both the forms and the query builder

**Review by lexiclevenger — APPROVED** [review]
### Testing instructions

- Customize the schema for collectionobjectgroup and collectionobjectgroupjoin
- Go to the query builder (/specify/query/new/collectionobjectgroup/) or to the form (/specify/view/collectionobjectgroup/new/)
- [x] Verify that the custom visibility/labels/descriptions are respected

---

Looks good! 👍

**Review by Areyes42 — APPROVED** [review]
### Testing instructions

-   Customize the schema for collectionobjectgroup and collectionobjectgroupjoin
-   Go to the query builder (/specify/query/new/collectionobjectgroup/) or to the form (/specify/view/collectionobjectgroup/new/)
-  [x]  Verify that the custom visibility/labels/descriptions are respected
***
Looks good!!

**Review by maxpatiiuk — COMMENTED** [review]
hmm, this looks like a back-end bug, maybe migration-related.
why is this being worked-around on the front-end?

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
@maxpatiiuk I was thinking the same thing. Isn't this something we should fix in the migrations?

**Comment by CarolineDenis** [comment]
@maxpatiiuk @grantfitzsimmons yes it will be done in: https://github.com/specify/specify7/pull/5221

File: specifyweb/frontend/js_src/lib/components/DataModel/specifyTable.ts [hunk 1]
```diff
@@ -241,9 +241,10 @@ export class SpecifyTable<SCHEMA extends AnySchema = AnySchema> {
     });
 
     const useLabels = getCache('forms', 'useFieldLabels') ?? true;
-    this.localization = getSchemaLocalization()[this.name.toLowerCase()] ?? {
-      items: {},
-    };
+    this.localization = getSchemaLocalization()[this.name.toLowerCase()] ??
+      getSchemaLocalization()[this.name] ?? {
+        items: {},
+      };
```
Comment (melton-jason) [inline specifyweb/frontend/js_src/lib/components/DataModel/specifyTable.ts:247]: If we do need a workaround on the frontend, would `caseInsensitiveHash` be fine here?
(Though I am mostly worried about the difference in performance with `caseInsensitiveHash` in this case, so the way it is currently is probably fine IF a workaround is needed). 

```suggestion
    this.localization = caseInsensitiveHash(
      getSchemaLocalization(),
      this.name
    ) ?? {
      items: {},
    };
```

**Review by melton-jason — COMMENTED** [review]
Can this be closed now that #5221 is merged? 
Or is the plan to include this frontend workaround regardless?

**Pull request closed without merging**